packages/c/commonlibsf/rules/plugin.lua

```diff
@@ -72,25 +79,59 @@ rule("plugin")
+            end
+            if struct then
+                configs.options.layout_dependent = false
+            else
+                configs.options.no_struct_use = struct
+                configs.options.layout_dependent = layout
+            end
+        else
+            configs.options = {
+                sig_scanning = false,
+                address_library = true,
+                no_struct_use = false,
+                layout_dependent = true
+            }
         end
 
         local config_map = {
-            PLUGIN_AUTHOR         = configs.author or "",
-            PLUGIN_DESCRIPTION    = configs.description or "",
-            PLUGIN_EMAIL          = configs.email or "",
-            PLUGIN_LICENSE        = (target:license() or "Unknown") .. " License",
-            PLUGIN_NAME           = configs.name or target:name(),
-            PLUGIN_VERSION        = target:version() or "0.0.0",
-            PLUGIN_VERSION_MAJOR  = semver.new(target:version() or "0.0.0"):major(),
-            PLUGIN_VERSION_MINOR  = semver.new(target:version() or "0.0.0"):minor(),
-            PLUGIN_VERSION_PATCH  = semver.new(target:version() or "0.0.0"):patch(),
-            PROJECT_NAME          = project.name() or "",
-            PROJECT_VERSION       = project.version() or "0.0.0",
-            PROJECT_VERSION_MAJOR = semver.new(project.version() or "0.0.0"):major(),
-            PROJECT_VERSION_MINOR = semver.new(project.version() or "0.0.0"):minor(),
-            PROJECT_VERSION_PATCH = semver.new(project.version() or "0.0.0"):patch(),
+            PLUGIN_AUTHOR           = configs.author or "",
+            PLUGIN_DESCRIPTION      = configs.description or "",
+            PLUGIN_EMAIL            = configs.email or "",
+            PLUGIN_LICENSE          = (target:license() or "Unknown") .. " License",
+            PLUGIN_NAME             = configs.name or target:name(),
+            PLUGIN_VERSION          = target:version() or "0.0.0",
+            PLUGIN_VERSION_MAJOR    = semver.new(target:version() or "0.0.0"):major(),
+            PLUGIN_VERSION_MINOR    = semver.new(target:version() or "0.0.0"):minor(),
+            PLUGIN_VERSION_PATCH    = semver.new(target:version() or "0.0.0"):patch(),
+            PROJECT_NAME            = project.name() or "",
+            PROJECT_VERSION         = project.version() or "0.0.0",
+            PROJECT_VERSION_MAJOR   = semver.new(project.version() or "0.0.0"):major(),
+            PROJECT_VERSION_MINOR   = semver.new(project.version() or "0.0.0"):minor(),
+            PROJECT_VERSION_PATCH   = semver.new(project.version() or "0.0.0"):patch(),
+            OPTION_SIG_SCANNING     = tostring(configs.options.sig_scanning),
+            OPTION_ADDRESS_LIBRARY  = tostring(configs.options.address_library),
+            OPTION_NO_STRUCT_USE    = tostring(configs.options.no_struct_use),
+            OPTION_LAYOUT_DEPENDENT = tostring(configs.options.layout_dependent)
```

Comment:
`tostring` shouldn't be needed here unless the config parser is failing?

Reply:
configs.options are designed to always have a correct true/false value, not nil, but I was not sure how Lua handles bool-to-string conversion. I can see you use
```
if type(result) ~= "string" then
    result = tostring(result)
end
```
next so tostring in config_map is not required indeed. If you set `OPTION_SIG_SCANNING = configs.options.sig_scanning` then `type(OPTION_SIG_SCANNING)` will be "boolean" and `OPTION_SIG_SCANNING = tostring(OPTION_SIG_SCANNING)` will be executed.

tostring in config_map is obsolete, yes.